### PR TITLE
Touch support

### PR DESCRIPTION
**Pull request by vxsx**
This PR adds generic touch support for modal, toolbar and structureboard drag and drop
touch support is not good enough everywhere yet, mostly because some things are very heavily hover based, but that will change soon.

Also fixes few bugs with drag n drop to empty placeholder and general jumping of the structureboard.
Also fixes annoying scrolling bug when you start dragging a plugin on a long page and it immediately scrolls all the way to the bottom

PR also contains style changes, at the moment it looks like this

screen shot 2015-07-22 at 17 22 09

Keep in mind that this PR is pointing to `preview` branch and not components are completely ready yet.

### CODE REVIEW

File: cms/static/cms/js/libs/pep.js [hunk 1]
```diff
@@ -0,0 +1,1330 @@
+/*!
```
Comment (FinalAngel) [inline cms/static/cms/js/libs/pep.js]: even though they are libs and in the end it gets packaged, I'll recommend adding it minified. Just to make sure that no code gets changed accidently.

File: cms/static/cms/js/modules/cms.modal.js [hunk 2]
```diff
@@ -344,9 +344,9 @@
 
                 this.ui.shim.show();
 
-                this.ui.body.on('mousemove.cms', function (e) {
-                    var left = position.left - (initial.pageX - e.pageX);
-                    var top = position.top - (initial.pageY - e.pageY);
+                this.ui.body.attr('touch-action', 'none').on('pointermove.cms', function (e) {
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.modal.js]: we still need to find a version to make `touch-action` either validate or getting a `data-touch-action` workaround

File: cms/static/cms/js/modules/cms.modal.js [hunk 3]
```diff
@@ -375,16 +375,17 @@
...
-                    var mvX = initial.pageX - e.pageX;
-                    var mvY = initial.pageY - e.pageY;
+                this.ui.body.attr('touch-action', 'none').on('pointermove.cms', function (e) {
+                    var mvX = initial.originalEvent.pageX - e.originalEvent.pageX;
+                    var mvY = initial.originalEvent.pageY - e.originalEvent.pageY;
 
                     var w = width - (mvX * 2);
                     var h = height - (mvY * 2);
-                    var max = 680;
+                    var wmax = 680;
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.modal.js]: wouldn't it be `wMax` ?

File: cms/static/cms/js/modules/cms.plugins.js [hunk 4]
```diff
@@ -147,7 +147,7 @@
                 this._setSubnav(draggable.find('> .cms-dragitem .cms-submenu'));
 
                 // adds event for hiding the subnav
-                draggable.bind('mouseenter mouseleave mouseover', function (e) {
+                draggable.bind('pointerenter pointerleave pointerover', function (e) {
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.plugins.js]: no cms namespace here?

File: cms/static/cms/js/modules/cms.plugins.js [hunk 5]
```diff
@@ -480,13 +480,13 @@
             _setSubnav: function (nav) {
                 var that = this;
 
-                nav.bind('mouseenter mouseleave tap.cms', function (e) {
+                nav.bind('pointerenter pointerleave tap.cms', function (e) {
                     e.preventDefault();
                     e.stopPropagation();
-                    (e.type === 'mouseenter') ? that._showSubnav($(this)) : that._hideSubnav($(this));
+                    (e.type === 'pointerenter') ? that._showSubnav($(this)) : that._hideSubnav($(this));
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.plugins.js]: also on all the ifs, namespacing?

File: cms/static/cms/js/modules/jquery.ui.touchpunch.js [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+ * jQuery UI Touch Punch 0.2.3
+ *
+ * Copyright 2011–2014, Dave Furfero
+ * Dual licensed under the MIT or GPL Version 2 licenses.
+ *
+ * Depends:
+ *  jquery.ui.widget.js
+ *  jquery.ui.mouse.js
+ */
+(function ($) {
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/jquery.ui.touchpunch.js]: I'd also minify this

File: cms/static/cms/sass/components/_structureboard.scss [hunk 7]
```diff
@@ -114,7 +117,7 @@
         padding-bottom: 0;
     }
     .cms-dragbar-empty-wrapper {
-        display: none;
+        /* display: none; */
```
Comment (FinalAngel) [inline cms/static/cms/sass/components/_structureboard.scss]: why not removing?

File: cms/static/cms/sass/components/_structureboard.scss [hunk 8]
```diff
@@ -126,7 +129,7 @@
     }
     .cms-draggables .cms-draggables {
         display: none;
-        min-height: 25px;
+        /* min-height: 25px; */
```
Comment (FinalAngel) [inline cms/static/cms/sass/components/_structureboard.scss]: also here: why not removing?

File: cms/templates/cms/toolbar/toolbar.html [hunk 9]
```diff
@@ -1,7 +1,7 @@
 {% load i18n l10n %}
 
 {% language request.toolbar.toolbar_language %}
-<div id="cms-toolbar" class="cms-reset{% if not user.is_authenticated %} cms-toolbar-auth{% endif %}{% if debug %} cms-toolbar-debug{% endif %}" dir="ltr">
+<div touch-action="none" id="cms-toolbar" class="cms-reset{% if not user.is_authenticated %} cms-toolbar-auth{% endif %}{% if debug %} cms-toolbar-debug{% endif %}" dir="ltr">
```
Comment (FinalAngel) [inline cms/templates/cms/toolbar/toolbar.html]: same on touch action notes

File: cms/static/cms/js/modules/cms.plugins.js (hunk 4 above)
Comment (vxsx) [inline cms/static/cms/js/modules/cms.plugins.js]: tried not to change much for the moment, only relevant parts. cleanup of this thing will follow, same way we did for modal

File: cms/static/cms/js/modules/cms.plugins.js (hunk 5 above)
Comment (vxsx) [inline cms/static/cms/js/modules/cms.plugins.js]: nope, here e.type is strictly the event name, without namespaces

**Comment by FinalAngel** [comment]
thanks for the updates :+1:

**Pull request merged**